velox/exec/Exchange.h

```diff
@@ -175,6 +175,7 @@ class ExchangeQueue {
 
   void closeLocked() {
     queue_.clear();
+    clearAllPromises();
```

Comment:
Are we safe to fulfill promises under the lock?

Reply:
They are fulfilled under the lock in other places.
For example:
```
  ~ExchangeQueue() {
    std::lock_guard<std::mutex> l(mutex_);
    clearAllPromises();
  }

```

Reply:
We should fix as
```
  ~ExchangeQueue() {
    promises;
    {
      std::lock_guard<std::mutex> l(mutex_);
      clearAllPromises(promises);
    }
    promises.set();
  }
```